Packaging-Roadmap-for-Mantl.md

```diff
@@ -0,0 +1,145 @@
+    - disable requiretty in sudoers -> sed 's/^.+requiretty$/# Defaults requiretty/' /etc/sudoers #but only last entry
+    - set selinux policy based on ansible defaults 
+    - disable firewalld -> here is a partial go implementation:
+```
+package main
+import (
+  "fmt"
+  "log"
+  "os/exec"
+)
+
+func main () {
+  // disable firewalld
+  out, err := exec.Command("systemctl disable firewalld").Output()
+  if err != nil{
+    log.Fatal(err)
+  }
+
+  // check state of firewalld
+  out, err := exec.Command("firewalld-cmd --state").Output()
+  if err != nil{
+    log.Fatal(err)
+  }
+  // if the state is NOT not running, disable has failed
+  if out != not running{
+    log.Fatal(err)
+    fmt.println ("Firewalld is not disabled.")
+  }
+}
+```
+  - Ansible users tasks
+    - configure members of wheel group for passwordless sudo -> `sed 's/^%wheel/%wheel ALL=(ALL) NOPASSWD: ALL/' /etc/sudoers`
+    - create enabled os users, based on `users` ansible var ([docs](http://mantl.readthedocs.org/en/latest/getting_started/ssh_users.html))
+    - set ssh key for `users`
+    - delete disabled `users`
+  - Ansible ssl tasks
+    - copy local path ssl/cacert.pem to remote server /etc/pki/ca-trust/source/anchors/cacert.pem; chown to root
+    - notify handler update-ca-trust -> `update-ca-trust`
+
+- lvm: **mantl-lvm**
+  - System Dependencies
+    - mantl-common
+    - device-mapper-libs
+    - lvm2
+  - Ansible facts
+    - defaults: lvm_volume_group_name=mantl, lvm_physical_device= different things depending upon provider
+    - volume groups list -> `vgscan`
+  - Ansible volume tasks when volume has not been created
+    - create volume group, based on ansib
+    - enable lvmetad service -> `systemctl enable lvm2-lvmetad 2>/dev/null && systemctl restart lvm2-lvmetad`
+  - Ansible main tasks: set group name to blank if physical device is blank
+
+- collectd: **mantl-collectd**
+  - PR#36
+- logrotate: *this one should be split up and managed by each package that needs it*
+- consul-template: *this is in the same boat as mantl-consul*
+- docker: **mantl-docker**
+  - Notes
+    - needs logrotate config from ansible role
+    - there's a bunch of stuff here for LVM.
+      - We could create separate package for it **mantl-docker-lvm**
+      - Or, we could include it in **mantl-lvm**
+  - same goes for the collectd docker plugin
+  - System Dependencies
+    - docker
+    - docker-selinux
+    - mantl-common
+- logstash: **mantl-logstash**
+- nginx: **mantl-nginx**
+  - System dependencies
+    - nginx
+    - mantl-common
+  - Ansible main tasks
+    - make tls directory -> `mkdir -p /etc/nginx/ssl && chmod 0700 /etc/nginx/ssl`
+    - deploy tls files
+    - encrypt admin password -> look at mantl PR#1058 for adding passwd to consul
+- consul: **mantl-consul**
+  - Question: could this be in **mantl-common**?
+- dnsmasq: **mantl-dnsmasq**
+  - PR#26
+
+*Roles for controls*
+- vault: **mantl-vault**
```

Comment:
So, it needs to be on control nodes only. Are there any other parts of the base config that are control nodes only? Then maybe we could roll that into a **mantl-control** package